python/cuml/neighbors/nearest_neighbors_mg.pyx

```diff
@@ -192,23 +192,24 @@ def _build_part_inputs(cuda_arr_ifaces,
 
     arr_ints = []
     for arr in cuda_arr_ifaces:
-        X_m, input_ptr, n_rows, n_cols, dtype = \
-            input_to_dev_array(arr, order="F",
-                               convert_to_dtype=(np.float32
-                                                 if convert_dtype
-                                                 else None),
-                               check_dtype=[np.float32])
+        X_m, n_rows, n_cols, dtype = \
+            input_to_cuml_array(arr, order="F",
+                                convert_to_dtype=(np.float32
+                                                  if convert_dtype
+                                                  else None),
+                                check_dtype=[np.float32])
+        input_ptr = X_m.ptr
         arr_ints.append({"obj": X_m,
                          "data": input_ptr,
                          "shape": (n_rows, n_cols)})
 
-    for rankSize in parts_to_ranks:
-        rank, size = rankSize
-        rsp = <RankSizePair*> malloc(sizeof(RankSizePair))
-        rsp.rank = <int>rank
-        rsp.size = <size_t>size
+    for rank in parts_to_ranks:
```

Comment:
Order is very important here- specifically, each tuple of `rank/size` is ordered to correspond to the order of input partitions, so that the output can match match the input. If this is now being passed in as a { `rank`: [size]} then we've lost the ordering of partitions wrt their associated ranks. 

I dont believe any change needs to be made to the `nearest_neighbors_mg` here, but rather the Dask wrapper should be building the proper format to pass into this function. You should be able to use `part_utils.parts_to_ranks` directly from the `distributed_data_handler.gpu_futures` for this.